gem/gem.py

```diff
@@ -149,18 +149,20 @@ def array(self):
 class Literal(Constant):
     """Tensor-valued constant"""
 
-    __slots__ = ('array',)
+    __slots__ = ('array', 'name')
     __front__ = ('array',)
+    __back__ = ('name',)
 
-    def __new__(cls, array):
+    def __new__(cls, array, name=None):
```

Comment:
In pre-evaluation, when I contract to form a new tensor, I give it a name such as "PE0".
Maybe this should be put together with the latex changes?